### DISCUSSION TITLE
Object Initializer

### DISCUSSION THREAD

**Discussion by MhAllan** [dotnet/csharplang#8277] [Language Ideas]
There is a need to validate instances created by initializers only, everytime I have domain model I find myself forced to create constructor with parameters just for the sake of validation, you can call validation int the property init or set, but that will be repeated a lot and will be called randomly while you want to validate after all properties are set, I suggest to have Initializer, which is called after all properties been set in object initialization. 

* Initializer should be called after object is created like this only: new Foo { Bar = .. };
* Initializer should return null object if it failed with exception

```csharp
class Foo
{
      public string Bar { get; init; }
      
     public init Foo()
     {
            //called after object is constructed using the initializer model
     }
}

**Comment by ufcpp** [comment]
See final initializers section in

https://github.com/dotnet/csharplang/blob/4fe69eb47ee40c878df4fa8897cc95655c6af824/meetings/2020/LDM-2020-05-04.md
https://github.com/dotnet/csharplang/blob/4fe69eb47ee40c878df4fa8897cc95655c6af824/meetings/2020/LDM-2020-04-27.md

**Comment by TahirAhmadov** [comment]
https://github.com/dotnet/csharplang/discussions/6591
